src/Core/ExceptionWrapper.php

```diff
@@ -31,16 +31,37 @@
 class ExceptionWrapper
 {
     /**
-     * Wrap Guzzle BadResponseException which returns truncated exception messages for 4xx and 5xx responses.
+     * Wrap Guzzle's ClientException which is thrown on 4xx response while http_errors=true
+     * Adds full response body to exception message
+     *
+     * @param ClientException $ex
+     * @return GraphRequestException
+     */
+    public static function wrapGuzzleClientException(\GuzzleHttp\Exception\ClientException $ex)
+    {
+        if ($ex->getResponse()) {
+            $errMsg = "Received {$ex->getResponse()->getStatusCode()} for call to {$ex->getRequest()->getUri()}\nAPI response: {$ex->getResponse()->getBody()->getContents()}";
+        } else {
+            $errMsg = $ex->getMessage();
+        }
+        return new GraphRequestException($errMsg, $ex);
+    }
+
+    /**
+     * Wrap Guzzle ServerException thrown on 5xx response while http_errors=true
      * Adds response body to the exception message.
      *
-     * @param BadResponseException $ex
-     * @return GraphException containing HTTP response from Graph API
+     * @param ServerException $ex
+     * @return GraphServerException containing HTTP response from Graph API
      * 
      */
-    public static function wrapGuzzleBadResponseException(BadResponseException $ex)
+    public static function wrapGuzzleServerException(\GuzzleHttp\Exception\ServerException $ex)
     {
-        $errMsg = "Received {$ex->getResponse()->getStatusCode()} for call to {$ex->getRequest()->getUri()}\nAPI response: {$ex->getResponse()->getBody()->getContents()}";
-        return new GraphException($errMsg);
+        if ($ex->getResponse()) {
+            $errMsg = "Received {$ex->getResponse()->getStatusCode()} for call to {$ex->getRequest()->getUri()}\nAPI response: {$ex->getResponse()->getBody()->getContents()}";
```

Comment:
Note that Guzzle obfuscates the possible user password to `***` in the URI for the message, to prevent security disclosure

Reply:
Also that's missing the response reason phrase (after the status code) and the request method (before the URI)